### PR TITLE
Snr rejection

### PR DESCRIPTION
**Pull request by talonmyburgh**
In this PR, we are replacing the residual delay threshold check with an SNR measure of the delay delta in the `ifft` of the gains to the surround (noise).

### CODE REVIEW

**Comment by talonmyburgh** [comment]
This branch has yet to be tested in production after the Voyager observation.

File: calibration_residual_kernals.py [hunk 1]
```diff
@@ -136,29 +139,27 @@ def calc_residuals_from_ifft(ant_to_gains, observation_frequencies, current_phas
...
+                        residual_delays[stream_idx] = 0.0
+                        phase_cals[stream_idx] = phase_matrix[stream_idx]
                     
                     #Calculating the spread of phases collected from each antennas from regions where we have actual gain values
-                    #sigma_phase[antenna_index, stream_idex] = np.std(phase_cals[stream_idx, collected_gain_range], axis = 1)
+                    sigma_phase[stream_idx] = np.std(phase_cals[stream_idx, frequency_indices[tune]], axis = 1)
 
             delay_residual_map[ant] = residual_delays
             phase_cal_map[ant] = phase_cals
+            snr_map[ant] = snr
```
Comment (talonmyburgh) [inline calibration_residual_kernals.py:161]: Do we want to return these dictionaries of `{ant : [snr]}` and `{ant : [sigma_phase]}` from the kernel for plotting in the slack channel? Is this a useful metric for visualisation at all - obviously outside of being a part of the rejection criteria...

Comment (jack-h) [reply calibration_residual_kernals.py:161]: I think this is definitely a useful metric to print out somewhere in slack.

**Comment by jack-h** [comment]
Has the SNR algorithm actually been tested on real data? I.e., does it give plausible SNRs on a known good data set and a known bad one? If so, this all looks fine, but let's not rock the boat till after today's Voyager.

**Comment by talonmyburgh** [comment]
No but this branch is ready to switch into production once the Voyager observation is passed (hopefully).

**Comment by jack-h** [comment]
OK. Wait till after Voyager, but this should be tested with some
pre-existing uvh5s if possible, rather than just waiting for observations
to come along and illuminate issues

**Comment by talonmyburgh** [comment]
In action - likely to continue to be edited but considered stable.

**Pull request merged**